log/log.go

```diff
@@ -72,7 +72,14 @@ type key int
 
 var logKey key
 
+// Append adds the given keyval pair to the given context. If the context already stores keyvals, the new ones get appended.
+// This should be used to track the logging context across the application.
+func Append(ctx context.Context, key, val any) context.Context {
```

Comment:
why not just rename AppendKeyVals to Append?

Reply:
both funcs are useful. `Append` makes sure the correct number of values is submitted, but sometimes we need to deal with slices

Reply:
convinced :)